zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkCallBuilder.java

```diff
@@ -78,12 +83,19 @@ public BulkCallBuilder(ElasticsearchStorage es, float esVersion, String tag) {
     if (pipeline != null) urlBuilder.addParam("pipeline", pipeline);
     if (waitForRefresh) urlBuilder.addParam("refresh", "wait_for");
 
-    Buffer okioBuffer = new Buffer();
-    for (IndexEntry<?> entry : entries) {
-      write(okioBuffer, entry, shouldAddType);
-    }
+    final HttpData body;
 
-    HttpData body = HttpData.wrap(okioBuffer.readByteArray());
+    CompositeByteBuf sink = RequestContext.mapCurrent(
```

Comment:
some comment for those less familiar with composite heap buffer etc? basically what does this buy us?

Reply:
Good idea.